setup.sh

```diff
@@ -96,14 +138,36 @@ apply () {
         else
                 $TERRAFORMPATH apply --var-file=$TERRAFORMTFVARS --auto-approve
         fi
+
+        # For development stacks, for the `infra-networking-route53` project, ensure that
+        # the shared_dev_route53 resource has been imported into terraform before applying
+        if [ $ENV != 'production' -a $ENV != 'staging' -a "$1" = 'infra-networking-route53' ] ; then
+                cd $ROOTPROJ
+                SEARCH="initialised = \"0\""
+                if grep -q "$SEARCH" "stacks/${ENV}.tfvars"; then
+                        import_shared_dev_route53
+                        create_stack_tfvars 1
+
+                        if [ "${USE_AWS_VAULT}" = 'true' ] ; then
+                                aws-vault exec ${PROFILE_NAME} -- $TERRAFORMPATH apply --var-file=$TERRAFORMTFVARS --auto-approve
+                        else
+                                $TERRAFORMPATH apply --var-file=$TERRAFORMTFVARS --auto-approve
+                        fi
+                fi
+        fi
 }
 
 destroy () {
+# Destroy a terraform project
         echo $1
 
         cd $TERRAFORMPROJ$1
         if [ "${USE_AWS_VAULT}" = 'true' ] ; then
                 aws-vault exec ${PROFILE_NAME} -- $TERRAFORMPATH destroy --var-file=$TERRAFORMTFVARS --auto-approve
+                # What is this doing?
```

Comment:
Sorry, @kentsanggds could you just explain to me what this logic is doing but more so why it is doing it please?

Reply:
Do you mean `168`, when destroying we need to ignore `infra-networking-route53` errors otherwise it will always interrupt the destroy all as we can't ignore the `shared_dev` resource, see here for more detail - https://github.com/alphagov/prometheus-aws-configuration-beta/pull/24/commits/a6b7435107b37934aca2e02a803cf45207ae2b05